test/hooks/test-hooks-index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var assert = require('assert');
+var Module = require('module');
+var findModuleVersion = require('../../lib/hooks/index.js').findModuleVersion;
+
+describe('findVersion', function() {
+  it('should correctly find package.json for userspace packages', function() {
+    assert.equal(findModuleVersion('express', module, Module._load), '4.13.3');
```

Comment:
We could use the versions file and check that we loaded a compatible version. Alternatively, we could have a separate package.json that loads precise versions of certain modules. That would require a separate npm install but it would be easier to keep it up to date in parallel with the test. What we really want is the ability to require a particular version of a module or require by github url but that functionality isn't available yet.